api/storage_service.cc

```diff
@@ -1011,6 +1011,25 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         }) | std::ranges::to<std::vector>();
     });
 
+    ss::get_object_store_endpoints.set(r, [&ctx](std::unique_ptr<http::request> req) -> future<json::json_return_type>  {
+        std::vector<ss::object_store_endpoint> endpoints;
+        auto& db = ctx.db.local();
```

Comment:
I try to discourage people from using any objects from `ctx`. All that we have in api/ code is so far legacy and is not to be inspired from

Reply:
Can you please point me to a non-legacy alternative to get a db handle?

Reply:
You don't need db here, you need db::config. So if you move the endpoint handler to api/config.cc, then you'll have it there